androidCore/impl/src/debug/kotlin/com/walletconnect/android/impl/di/StorageModule.kt

```diff
@@ -11,17 +12,20 @@ fun coreStorageModule() = module {
 
     includes(baseStorageModule())
 
-    single<SqlDriver>(named(AndroidCoreDITags.ANDROID_CORE_DATABASE_DRIVER)) {
-        AndroidSqliteDriver(
-            schema = AndroidCoreDatabase.Schema,
-            context = androidContext(),
-            name = DBNames.ANDROID_CORE_DB_NAME,
-        )
+    if (wcKoinApp.koin.getOrNull<SqlDriver>(named(AndroidCoreDITags.ANDROID_CORE_DATABASE_DRIVER)) == null) {
+        single<SqlDriver>(named(AndroidCoreDITags.ANDROID_CORE_DATABASE_DRIVER)) {
+            AndroidSqliteDriver(
+                schema = AndroidCoreDatabase.Schema,
+                context = androidContext(),
+                name = DBNames.ANDROID_CORE_DB_NAME,
+            )
+        }
+    } else {
+        wcKoinApp.koin.get<SqlDriver>(named(AndroidCoreDITags.ANDROID_CORE_DATABASE_DRIVER))
     }
```

Comment:
This can maybe also be done so that we only do the get call once.

```suggestion
   wcKoinApp.koin.getOrNull<SqlDriver>(named(AndroidCoreDITags.ANDROID_CORE_DATABASE_DRIVER) 
       ?: single<SqlDriver>(named(AndroidCoreDITags.ANDROID_CORE_DATABASE_DRIVER)) {
            AndroidSqliteDriver(
                schema = AndroidCoreDatabase.Schema,
                context = androidContext(),
                name = DBNames.ANDROID_CORE_DB_NAME,
            )
        }
    
```

Reply:
I replaced all places